IGC/OCLFE/igd_fcl_mcl/source/clang_tb.cpp

```diff
@@ -771,34 +771,49 @@ namespace TC
 
         const char * optValue = optSubstring + optName.size();
         const char * end = optValue + strlen(optValue);
+        std::string_view opt(optValue, end - optValue);
 
         // parse
-        const std::string invalidFormatMessage = "Invalid format of -cl-std option, expected -cl-std=CLMAJOR.MINOR";
-        auto isNumeric = [](char v) { return (v >= '0') && (v <= '9'); };
-        if (false == ((end - optValue >= 5) && (optValue[0] == 'C') && (optValue[1] == 'L') && isNumeric(optValue[2])
-            && (optValue[3] == '.') && isNumeric(optValue[4])
-            )
-            ) {
-            exceptString = invalidFormatMessage;
-            return 0;
-        }
-
         unsigned int retVersion = 0;
-        // subverions
-        if ((end - optValue >= 7) && (optValue[5] != ' ')) {
-            if ((optValue[5] == '.') || isNumeric(optValue[6])) {
-                retVersion += optValue[6] - '0';
+        if (opt.find("CLC++") == 0) {
```

Comment:
Here we should define `invalidFormatMessage` with appropriate msg for CLC++
```suggestion
        if (opt.find("CLC++") == 0) {
        const std::string invalidFormatMessage = "Invalid format of -cl-std option, expected -cl-std=CLC++, -cl-std=CLC++1.0, or -cl-std=CLC++2021";
```